test/clear-kvm.make

```diff
@@ -71,14 +68,19 @@ NUM_NODES = 4
 # kubernetes-0/1/2/.... The first image is for the Kubernetes master node,
 # but configured so that also normal apps can run on it, i.e. no additional
 # worker nodes are needed.
-_work/clear-kvm.img: test/setup-clear-kvm.sh _work/clear-kvm-original.img _work/kube-clear-kvm _work/OVMF.fd _work/start-clear-kvm _work/id _work/passwd
-	$(PROXY_ENV) test/setup-clear-kvm.sh $(NUM_NODES)
-	ln -sf clear-kvm.0.img $@
+_work/$(CLUSTER)/created: _work/clear-kvm%/created: test/setup-clear-kvm.sh _work/clear-kvm%/start-kubernetes _work/clear-kvm%/run-qemu _work/id _work/passwd
```

Comment:
My original approach was just a wildcard rule:
```
_work/clear-kvm%/created: test/setup-clear-kvm.sh _work/clear-kvm%/start-kubernetes _work/clear-kvm%/run-qemu _work/id _work/passwd
```

But then `make CLUSTER=clear-kvm`, only `make CLUSTER=clear-kvm-something` did. Apparently make absolutely wanted the `%` to be non-empty.

The way it is written now currently works for clusters called "clear-kvm[-something]" but it'll fail for other cluster names:
```
$ make CLUSTER=govm-foobar start
test/clear-kvm.make:71: target '_work/govm-foobar/created' doesn't match the target pattern
test/clear-kvm.make:80: target '_work/govm-foobar/run-qemu' doesn't match the target pattern
test/clear-kvm.make:112: target '_work/govm-foobar/start-kubernetes' doesn't match the target pattern
if ! [ -f _work/clear-kvm-28920.img ]; then ( true && mkdir -p _work && cd _work && version=28920 && [ "$version" ] && curl -O https://download.clearlinux.org/releases/$version/clear/clear-$version-kvm.img.xz && curl -O https://download.clearlinux.org/releases/$version/clear/clear-$version-kvm.img.xz-SHA512SUMS && curl -O https://download.clearlinux.org/releases/$version/clear/clear-$version-kvm.img.xz-SHA512SUMS.sig && sed -e 's;/.*/;;' clear-$version-kvm.img.xz-SHA512SUMS | sha512sum -c && unxz -c <clear-$version-kvm.img.xz >clear-kvm-$version.img ); fi
```

This is a problem that'll have to be solved when adding some alternative setup mechanism.

Reply:
Found a solution, see the second commit.

With, it becomes possible to add `clear-govm` like this (no implementation yet...):
```
diff --git a/Makefile b/Makefile
index 1e3e053f..276561ed 100644
--- a/Makefile
+++ b/Makefile
@@ -81,6 +81,7 @@ clean:
 # same IP addresses, and thus cannot run in parallel.
 CLUSTER := clear-kvm
 include test/clear-kvm.make
+include test/clear-govm.make
 include test/start-stop.make
 include test/test.make
 
diff --git a/test/clear-govm.make b/test/clear-govm.make
new file mode 100644
index 00000000..f2e877cd
--- /dev/null
+++ b/test/clear-govm.make
@@ -0,0 +1,8 @@
+# The following rules only apply when CLUSTER starts with clear-govm.
+ifneq (,$(filter clear-govm%,$(CLUSTER)))
+
+_work/$(CLUSTER)/created: _work/clear-govm%/created:
+       echo "not implemented yet"
+       false
+
+endif
```